conda/pytorch-nightly/meta.yaml

```
@@ -1,68 +1,92 @@
   version: "{{ environ.get('PYTORCH_BUILD_VERSION') }}"
 
 source:
   path: "{{ environ.get('PYTORCH_GITHUB_ROOT_DIR') }}"
 
-requirements:
-  build:
-    - cmake
-    - {{ compiler('c') }} # [win]
-
-  host:
-    - python
-    - numpy >=1.11.*
-    - setuptools
-    - pyyaml
-    - mkl >=2019
-    - mkl-include
-    - typing_extensions
-    - dataclasses # [py36]
-    - ninja
-    - libuv # [win]
-    - libuv # [unix]
-    - pkg-config # [unix]
-{{ environ.get('MAGMA_PACKAGE') }}
 
-  run:
-    - python
-    - numpy >=1.11
-    - mkl >=2018
-    - dataclasses # [py36]
-    - ninja
-    - typing_extensions
-    - blas * mkl
-{{ environ.get('CONDA_CUDATOOLKIT_CONSTRAINT') }}
+outputs:
+  # The actual pytorch package
+  - name: pytorch
+    version: "{{ environ.get('PYTORCH_BUILD_VERSION') }}"
+    build:
+      number: {{ environ.get('PYTORCH_BUILD_NUMBER') }}
+      detect_binary_files_with_prefix: False
+      string: "{{ environ.get('PYTORCH_BUILD_STRING') }}"
+      script_env:
+        - CUDA_VERSION
+        - CUDNN_VERSION
+        - USE_CUDA
+        - CMAKE_ARGS
+        - EXTRA_CAFFE2_CMAKE_FLAGS
+        - DEVELOPER_DIR
+        - DEBUG
+        - USE_FBGEMM
+        - USE_SCCACHE  # [win]
+        - USE_DISTRIBUTED  # [unix]
 
-build:
-  number: {{ environ.get('PYTORCH_BUILD_NUMBER') }}
-  detect_binary_files_with_prefix: False
-  string: "{{ environ.get('PYTORCH_BUILD_STRING') }}"
-  script_env:
-    - CUDA_VERSION
-    - CUDNN_VERSION
-    - CONDA_CUDATOOLKIT_CONSTRAINT
-    - USE_CUDA
-    - CMAKE_ARGS
-    - EXTRA_CAFFE2_CMAKE_FLAGS
-    - DEVELOPER_DIR
-    - DEBUG
-    - USE_FBGEMM
-    - USE_SCCACHE # [win]
-    - USE_DISTRIBUTED # [unix]
-  features:
-{{ environ.get('CONDA_CPU_ONLY_FEATURE') }}
+    requirements:
+      build:
+        - cmake
+        - {{ compiler('c') }} # [win]
+      host:
+        - python
+        - numpy 1.11.*
+        - setuptools
+        - pyyaml
+        - mkl >=2019
+        - mkl-include
+        - typing_extensions
+        - dataclasses # [py36]
+        - ninja
+        - libuv
+        - pkg-config  # [unix]
+{% if desired_cuda != "cpu" %}
+        - magma-cuda{{ cuda_maj_min_nodot }}  # [not osx and not win]
+{% endif %}
 
-test:
- imports:
-    - torch
- source_files:
-   - test
- commands:
-   - OMP_NUM_THREADS=4 python ./test/run_test.py || true # [not win]
-   - python ./test/run_test.py # [win]
+      run:
+        - python
+        - numpy >=1.11
+        - mkl >=2018
+        - ninja
+        - typing_extensions
+        - blas * mkl
+        - gpu * {{ desired_cuda }}
+        - {{ pin_subpackage("cuda" + cuda_maj_min_nodot, exact=True) }}
 
+    test:
+      imports:
+        - torch
+      source_files:
+        - test
+      commands:
+        - OMP_NUM_THREADS=4 python ./test/run_test.py || true  # [not win]
+        - python ./test/run_test.py  # [win]
+  # mutex package for only one GPU type
+  - name: gpu
+    version: 1.0
+    build:
+      string: {{ desired_cuda }}
```

```suggestion
      string: "{{ desired_cuda }}"
```

Have to surround with quotes here otherwise it gets interpreted as an int